articles/tutorials/advanced/2d_shaders/02_hot_reload/index.md

```diff
@@ -0,0 +1,280 @@
+
+### Extending Content Manager
+
+Currently, the `grayscaleEffect.fx` is being loaded in the `GameScene` 's `LoadContent()` method like this:
+
+[!code-csharp[](./snippets/snippet-2-16.cs)]
+
+The `.Load()` function in the existing `ContentManager` is almost sufficient for our needs, but it returns a regular `Effect`, which has no understanding of the dynamic nature of the new content workflow. 
+
+Create a new _Content_ folder within the _MonoGameLibrary_ project, add a new file named `ContentManagerExtensions.cs`, and add the following code for the foundation of the new system:
+
+[!code-csharp[](./snippets/snippet-2-17.cs)]
+
+Now, you can create an [extension method](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods) for the existing `MonoGame`'s `ContentManager` class:
+
+[!code-csharp[](./snippets/snippet-2-18.cs)]
+
+This new `Watch` function is an opportunity to enhance how content is loaded. Use this new function to load the `_greyscaleEffect` effect in the `GameScene`:
+
+[!code-csharp[](./snippets/snippet-2-19.cs)]
+
+
+### The `WatchedAsset` class
+
+The new system will need to keep track of additional information for each asset that we plan to be _hot-reloadable_. The new data will live in a new class, `WatchedAsset<T>`. 
+
+Add a new file named `WatchedAsset.cs` in the _MonoGameLibrary/Content_ folder:
+
+[!code-csharp[](./snippets/snippet-2-20.cs)]
+
+The new `Watch` method should return a `WatchedAsset<T>` instead of the direct `Effect`:
+
+[!code-csharp[](./snippets/snippet-2-21.cs)]
+
+This will require that the type of `_greyscaleEffect` change to a `WatchedAsset<Effect>` instead of simply an `Effect`:
+
+[!code-csharp[](./snippets/snippet-2-22.cs)]
+
+And that will cause a few compilation errors where the `_greyscaleEffect` is used throughout the rest of the `GameScene`. 
+The compile errors appear because `_greyscaleEffect` used to be an `Effect`, but now the `Effect` is actually available as `_grayscaleEffect.Asset`. 
+
+### Reload Extension
+
+It is time to write the extension method that the game code will use to opt into reloading an asset. From the earlier section, anytime a `.fx` file is updated, the compiled `.xnb` file will be copied into the game's runtime folder. The operating system will keep track of the last time the `.xnb` file was written, and we can leverage that information with the `WatchedAsset<T>.UpdatedAt` property to understand if the `.xnb` file is _newer_ than the current loaded `Effect`. 
+
+This method will take a `WatchedAsset<T>` and update the inner `Asset` property _if_ the `.xnb` file is newer. The method returns `true` when the asset is reloaded, which will be useful later:
+
+[!code-csharp[](./snippets/snippet-2-23.cs)]
+
+
+At the top of the `GameScene.Update()` method, add the following line to opt into reloading the `_grayscaleEffect` asset:
+
+[!code-csharp[](./snippets/snippet-2-24.cs?highlight=3-4)]
```

Comment:
Code is inverted between the sample in source and this snipped ( lines are reversed) please ensure they are consistant between sample and tutorial

Reply:
I will fix